### ISSUE TITLE
Freqtrade doesn't buy on dry-run mod

### ISSUE THREAD

**Issue by yigitserce** [freqtrade/freqtrade#4744]
freqtrade trade -c config.json --strategy NewStrategy
i've tried backtesting and it worked. 
when run this code , bot works but it doesn't buy or sell 

### Output
> 2021-04-15 23:56:07,204 - freqtrade.worker - INFO - Starting worker develop-b0bd7327
2021-04-15 23:56:07,204 - freqtrade.configuration.configuration - INFO - Using config: config.json ...
2021-04-15 23:56:07,205 - freqtrade.loggers - INFO - Verbosity set to 0
2021-04-15 23:56:07,205 - freqtrade.configuration.configuration - INFO - Runmode set to dry_run.
2021-04-15 23:56:07,205 - freqtrade.configuration.configuration - INFO - Dry run is enabled
2021-04-15 23:56:07,205 - freqtrade.configuration.configuration - INFO - Using DB: "sqlite:///tradesv3.dryrun.sqlite"
2021-04-15 23:56:07,205 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 15 ...
2021-04-15 23:56:07,205 - freqtrade.configuration.configuration - INFO - Using user-data directory: /home/perse/freqtrade/user_data ...
2021-04-15 23:56:07,206 - freqtrade.configuration.configuration - INFO - Using data directory: /home/perse/freqtrade/user_data/data/binance ...
2021-04-15 23:56:07,206 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2021-04-15 23:56:07,211 - freqtrade.configuration.check_exchange - INFO - Exchange "binance" is officially supported by the Freqtrade development team.
2021-04-15 23:56:07,211 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2021-04-15 23:56:07,212 - freqtrade.freqtradebot - INFO - Starting freqtrade develop-b0bd7327
2021-04-15 23:56:07,221 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy NewStrategy from '/home/perse/freqtrade/user_data/strategies/NewStrategy.py'...
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'timeframe' with value in config file: 5m.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: BTC.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 0.001.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'buy': 10, 'sell': 30}.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'use_sell_signal' with value in config file: True.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'sell_profit_only' with value in config file: False.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ignore_roi_if_buy_signal' with value in config file: False.
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'60': 0.01, '30': 0.02, '0': 0.04}
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 5m
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: False
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: False
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False}
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: BTC
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 0.001
2021-04-15 23:56:07,222 - freqtrade.
resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 30
2021-04-15 23:56:07,222 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'buy': 10, 'sell': 30}
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: False
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: False
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_offset: 0.0
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
2021-04-15 23:56:07,223 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
2021-04-15 23:56:07,223 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2021-04-15 23:56:07,225 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2021-04-15 23:56:07,225 - freqtrade.exchange.exchange - INFO - Using CCXT 1.47.47
2021-04-15 23:56:07,225 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True}
2021-04-15 23:56:07,231 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True, 'rateLimit': 200}
2021-04-15 23:56:07,236 - freqtrade.exchange.exchange - INFO - Using Exchange "Binance"
2021-04-15 23:56:09,560 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Binance'...
2021-04-15 23:56:09,662 - freqtrade.wallets - INFO - Wallets synced.
2021-04-15 23:56:09,664 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from '/home/perse/freqtrade/freqtrade/plugins/pairlist/StaticPairList.py'...
2021-04-15 23:56:09,665 - freqtrade.plugins.protectionmanager - INFO - No protection Handlers defined.
2021-04-15 23:56:09,945 - freqtrade.rpc.rpc_manager - INFO - Enabling rpc.telegram ...
2021-04-15 23:56:10,023 - apscheduler.scheduler - INFO - Scheduler started
2021-04-15 23:56:10,248 - freqtrade.rpc.telegram - INFO - rpc.telegram is listening for following commands: [['status'], ['profit'], ['balance'], ['start'], ['stop'], ['forcesell'], ['forcebuy'], ['trades'], ['delete'], ['performance'], ['stats'], ['daily'], ['count'], ['locks'], ['unlock', 'delete_locks'], ['reload_config', 'reload_conf'], ['show_config', 'show_conf'], ['stopbuy'], ['whitelist'], ['blacklist'], ['logs'], ['edge'], ['help'], ['version']]
2021-04-15 23:56:10,248 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'running'}
2021-04-15 23:56:10,508 - freqtrade.worker - INFO - Changing state to: RUNNING
2021-04-15 23:56:10,508 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': warning, 'status': 'Dry run is enabled. All trades are simulated.'}
2021-04-15 23:56:10,640 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "*Exchange:* `binance`\n*Stake per trade:* `0.001 BTC`\n*Minimum ROI:* `{'60': 0.01, '30': 0.02, '0': 0.04}`\n*Stoploss:* `-0.1`\n*Timeframe:* `5m`\n*Strategy:* `NewStrategy`"}
2021-04-15 23:56:10,765 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "Searching for BTC pairs to buy and sell based on [{'StaticPairList': 'StaticPairList'}]"}
2021-04-15 23:56:14,010 - numexpr.utils - INFO - NumExpr defaulting to 6 threads.
2021-04-15 23:56:15,871 - freqtrade.worker - INFO - Bot heartbeat. PID=6583, version='develop-b0bd7327', state='RUNNING'
2021-04-15 23:57:15,923 - freqtrade.worker - INFO - Bot heartbeat. PID=6583, version='develop-b0bd7327', state='RUNNING'

**Comment by Th0masL** [comment]
It's probably because your strategy is containing some "invalid" tests that only works in backtest.

For example, in backtest it's technically possible to "look in the future" (because all the data has already been downloaded and is known), using things like `dataframe['close'].shift(-1)`, which would not work in dry-run/live.

Please review your strategy, but if you want more help, you will have to share some parts of your strategy so we can help your troubleshooting more precisely.

**Comment by xmatthias** [comment]
With the information above, i'd also say that [this faq entry](https://www.freqtrade.io/en/stable/faq/#i-have-waited-5-minutes-why-hasnt-the-bot-made-any-trades-yet) might apply (on top of the above answer).

Some strategies trade many trades per day, others only a few - or have "downtimes" of multiple days, before then doing a lot of trading the other week.

**Comment by yigitserce** [comment]
> It's probably because your strategy is containing some "invalid" tests that only works in backtest.
> 
> For example, in backtest it's technically possible to "look in the future" (because all the data has already been downloaded and is known), using things like `dataframe['close'].shift(-1)`, which would not work in dry-run/live.
> 
> Please review your strategy, but if you want more help, you will have to share some parts of your strategy so we can help your troubleshooting more precisely.
``

> ### I directly implemented my old version strategy to new version freqtrade. Honestly i don't really know how to configure code for new version

>  **This is the Strategy**



import numpy as np  # noqa
import pandas as pd  # noqa
from pandas import DataFrame

from freqtrade.strategy import IStrategy
from freqtrade.strategy import CategoricalParameter, DecimalParameter, IntParameter

import talib.abstract as ta
import freqtrade.vendor.qtpylib.indicators as qtpylib

class BBRSI(IStrategy):

    # Strategy interface version - allow new iterations of the strategy interface.
    # Check the documentation or the Sample strategy to get the latest version.
    INTERFACE_VERSION = 2

    # Minimal ROI designed for the strategy.
    # This attribute will be overridden if the config file contains "minimal_roi".
    minimal_roi = {
     "0": 0.14055923307758864,
     "31": 0.027071866212491056,
     "82": 0.012758371821620255,
     "201": 0
    }

    # Optimal stoploss designed for the strategy.
    # This attribute will be overridden if the config file contains "stoploss".
    stoploss = -0.27203348102754255

    # Trailing stoploss
    trailing_stop = False
    trailing_only_offset_is_reached = False
    trailing_stop_positive = 0.01
    trailing_stop_positive_offset = 0.0  # Disabled / not configured

    # Optimal timeframe for the strategy.
    timeframe = '5m'

    # Run "populate_indicators()" only for new candle.
    process_only_new_candles = False

    # These values can be overridden in the "ask_strategy" section in the config.
    use_sell_signal = True
    sell_profit_only = False
    ignore_roi_if_buy_signal = True

    # Number of candles the strategy requires before producing valid signals
    startup_candle_count: int = 30

    # Optional order type mapping.
    order_types = {
        'buy': 'limit',
        'sell': 'limit',
        'stoploss': 'limit',
        'stoploss_on_exchange': False
    }

    # Optional order time in force.
    order_time_in_force = {
        'buy': 'gtc',
        'sell': 'gtc'
    }


    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Adds several different TA indicators to the given DataFrame

        Performance Note: For the best performance be frugal on the number of indicators
        you are using. Let uncomment only the indicator you are using in your strategies
        or your hyperopt configuration, otherwise you will waste your memory and CPU usage.
        :param dataframe: Dataframe with data from the exchange
        :param metadata: Additional information, like the currently traded pair
        :return: a Dataframe with all mandatory indicators for the strategies
        """

        # RSI
        dataframe['rsi'] = ta.RSI(dataframe)

        # Bollinger Bands
        bollinger = qtpylib.bollinger_bands(qtpylib.typical_price(dataframe), window=20, stds=2)
        dataframe['bb_lowerband'] = bollinger['lower']
        dataframe['bb_middleband'] = bollinger['mid']
        dataframe['bb_upperband'] = bollinger['upper']


        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Based on TA indicators, populates the buy signal for the given dataframe
        :param dataframe: DataFrame populated with indicators
        :param metadata: Additional information, like the currently traded pair
        :return: DataFrame with buy column
        """
        dataframe.loc[
            (
                (dataframe['rsi'] < 34) &
                (dataframe["close"] < dataframe['bb_lowerband'] )
            ),
            'buy'] = 1

        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Based on TA indicators, populates the sell signal for the given dataframe
        :param dataframe: DataFrame populated with indicators
        :param metadata: Additional information, like the currently traded pair
        :return: DataFrame with buy column
        """
        dataframe.loc[
            (
                (dataframe['rsi'] > 65) &
                (dataframe["close"] > dataframe['bb_upperband'] )
            ),
            'sell'] = 1

        return dataframe

**Comment by yigitserce** [comment]
> With the information above, i'd also say that [this faq entry](https://www.freqtrade.io/en/stable/faq/#i-have-waited-5-minutes-why-hasnt-the-bot-made-any-trades-yet) might apply (on top of the above answer).
> 
> Some strategies trade many trades per day, others only a few - or have "downtimes" of multiple days, before then doing a lot of trading the other week.

>Yes you are right. it is not code problem. it doesn't find buy signal but i solved it by using VolumePairList so now it check more coin and find more buy signal. 
_**Thank you**_
